src/lib/operators/aggregate_sort.cpp

```diff
@@ -0,0 +1,416 @@
+                                      AggregateFunctor<ColumnType, AggregateType> aggregate_function,
+                                      std::shared_ptr<const Table> sorted_table) {
+  const size_t num_groups = aggregate_group_offsets.size() + 1;
+  auto aggregate_results = std::vector<AggregateType>(num_groups);
+  auto aggregate_null_values = std::vector<bool>(num_groups, false);
+
+  uint64_t value_count = 0u;
+  uint64_t value_count_with_null = 0u;
+  std::unordered_set<ColumnType> unique_values;
+  uint64_t aggregate_group_index = 0u;
+
+  const auto& chunks = sorted_table->chunks();
+
+  std::optional<AggregateType> current_aggregate_value;
+  ChunkID current_chunk_id{0};
+  // Count*
+  if (function == AggregateFunction::Count && !_aggregates[aggregate_index].column) {
+    auto previous_group_offset = RowID{ChunkID{0u}, ChunkOffset{0}};
+    for (const auto &aggregate_group_offset : aggregate_group_offsets) {
+      if (previous_group_offset.chunk_id == aggregate_group_offset.chunk_id) {
+        value_count_with_null = aggregate_group_offset.chunk_offset - previous_group_offset.chunk_offset;
+      } else {
+        uint64_t count = 0;
+        count += chunks[previous_group_offset.chunk_id]->size() - previous_group_offset.chunk_offset;
+        for (auto chunk_id = previous_group_offset.chunk_id + 1; chunk_id < aggregate_group_offset.chunk_id;
+             chunk_id++) {
+          count += chunks[chunk_id]->size();
+        }
+        count += aggregate_group_offset.chunk_offset + 1;
+        value_count_with_null = count;
+      }
+      _set_and_write_aggregate_value<ColumnType, AggregateType, function>(
+          aggregate_results, aggregate_null_values, aggregate_group_index, aggregate_index, current_aggregate_value,
+          value_count, value_count_with_null, unique_values);
+      previous_group_offset = aggregate_group_offset;
+      aggregate_group_index++;
+    }
+    value_count_with_null = chunks[previous_group_offset.chunk_id]->size() - previous_group_offset.chunk_offset;
+    for (size_t chunk_id = previous_group_offset.chunk_id + 1; chunk_id < chunks.size(); chunk_id++) {
+      value_count_with_null += chunks[chunk_id]->size();
+    }
+
+  } else {
+    auto aggregate_group_offset_iter = aggregate_group_offsets.begin();
+    const auto column_id = _aggregates[aggregate_index];
+    const auto column = column_id.column;
+    for (const auto &chunk : chunks) {
+      auto segment = chunk->get_segment(*column);
+      segment_iterate<ColumnType>(*segment, [&](const auto& position) {
+        const auto row_id = RowID{current_chunk_id, position.chunk_offset()};
+        const auto is_new_group = !aggregate_group_offsets.empty() && row_id == *aggregate_group_offset_iter;
+        auto new_value = position.value();
+        if (is_new_group) {
+          _set_and_write_aggregate_value<ColumnType, AggregateType, function>(
```

Comment:
I'm not really sure what you mean. We do need to update the current aggregate value with each single row. How could we otherwise get the correct result for a e.g. sum?
Average is effectively a SUM, we do not calculate the average with each new incoming row.
However, the lines you marked are doing something different: They are outputting the aggregate value of the previous group to the result table. This happens (and must happen) every time we reach a new group (see L. 78).

Reply:
Is this still an issue @Miroka96?